extensions/amp-story/0.1/navigation-state.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {dev} from '../../../src/log';
+import {Observable} from '../../../src/observable';
+
+
+/**
+ * Types of state changes that can be consumed.
+ * @enum {number}
+ */
+export const StateChangeType = {
+  ACTIVE_PAGE: 0,
+};
+
+
+/** @typedef {{type: !StateChangeType, value: *}} */
```

Comment:
It'd be great if we could type the `value` key here with either several `value` typedefs or event typedefs. Let's address this later though. 😄

Reply:
Yeah, not super important right now :)